IframeMediaPause/iframe_media_pausing.md

```diff
@@ -0,0 +1,199 @@
+
+```js
+// AudioContext being create in a not rendered iframe, where
+// media-playback-while-not-rendered is not allowed. 
+let audioCtx = new AudioContext();
+let oscillator = audioCtx.createOscillator();
+oscillator.connect(audioCtx.destination);
+
+const resume_button = document.querySelector("resume_button");
+resume_button.addEventListener('click', () => {
+  if (audioCtx.state === "suspended") {
+    audioCtx.resume().then(() => {
+      console.log("Context resumed");
+    });
+  }
+})
+
+oscillator.start(0);
+// should print 'suspended'
+console.log(audioCtx.state)
+```
+
+Similarly, for scenario 2, when the iframe is not rendered, the audio context state should change to `'suspended'` and the website can monitor this by listening to the [`statechange`](https://webaudio.github.io/web-audio-api/#eventdef-baseaudiocontext-statechange) event. Then, when the iframe is rendered again, it should wait for a new user interaction, like in the first scenario, to resume playback.
+
+### Web Speech API
+
+The [Web Speech API](https://wicg.github.io/speech-api/) proposes a [SpeechSynthesis interface](https://wicg.github.io/speech-api/#tts-section). The latter interface allows websites to create text-to-speech output by calling [`window.speechSynthesis.speak`](https://wicg.github.io/speech-api/#dom-speechsynthesis-speak) with a [`SpeechSynthesisUtterance`](https://wicg.github.io/speech-api/#speechsynthesisutterance), which represents the text-to-be-said.
+
+For both scenarios, the iframe should listen for utterance errors when calling `window.speechSynthesis.speak`. For scenario 1 it should fail with a [`"not-allowed"`](https://wicg.github.io/speech-api/#dom-speechsynthesiserrorcode-not-allowed) SpeechSyntesis error; and, for scenario 2, it should fail with an [`"interrupted"`](https://wicg.github.io/speech-api/#dom-speechsynthesiserrorcode-interrupted) error.
+
+```js
+let utterance = new SpeechSynthesisUtterance('blabla');
+
+utterance.addEventListener('error', (event) => {
+  if (event.error === "not-allowed") {
+    console.log("iframe is not rendered yet");
+  } else if (event.error === "interrupted") {
+    console.log("iframe was hidden during speak call");
+  }
+})
+
+window.speechSynthesis.speak(utterance);
+```
+
+### Interoperability with autoplay
+
+If an iframe with the permission policy property set to `allow="media-playback-while-not-rendered 'none'; autoplay *"` stops being rendered, the `media-playback-while-not-rendered` permission policy should take precedence and no media should be played while it remains not rendered. The same behavior should happen if the [`autoplay`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/audio#autoplay) property has been set in a HTMLMediaElement 
```

Comment:
If the autoplay property is set to true for a media element and it is inside an iframe that goes from not rendered to rendered, should the media element start playing automatically?

Reply:
No. In this first proposal, we think that the pausing that happens because the iframe is not anymore rendered should be equivalent to an user manually pausing playback. So, to resume it, there must be an user gesture again.

Reply:
Re: user gesture

If the iframe has autoplay permission either through the autoplay permission policy or the user agent, then no user gesture should be required after the iframe's rendering resumes.  The iframe should be able to call play() outside of a user gesture to restart playback.

Reply:
I will create a follow up to PR to make this clearer.